### PR TITLE
community: update inclusive language guidelines

### PR DESCRIPTION
**Pull request by mattklein123**
Also fix some easy low hanging fruit.

Part of https://github.com/envoyproxy/envoy/issues/11596

I will follow up with additional targeted changes for redis and client_ssl_auth.

Risk Level: Low
Testing: Existing tests
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by mattklein123** [comment]
cc @envoyproxy/maintainers

File: source/extensions/tracers/xray/tracer.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class Span : public Tracing::Span, Logger::Loggable<Logger::Id::config> {
 
   /**
    * Adds a key-value pair to either the Span's annotations or metadata.
-   * A whitelist of keys are added to the annotations, everything else is added to the metadata.
+   * A allowlist of keys are added to the annotations, everything else is added to the metadata.
```
Comment (asraa) [inline source/extensions/tracers/xray/tracer.h:68]: nit: An

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,6 +21,23 @@ maximize the chances of your PR being merged.
 
 * See [STYLE.md](STYLE.md)
 
+# Inclusive language policy
+
+The Envoy community has an explicit goal to be inclusive to all. As such, all PRs must adhere to the
+following guidelines for all code, APIs, and documentation:
+
+* The following words and phrases are not allowed:
+  * *Whitelist*: use allowlist instead.
```
Comment (htuch) [inline CONTRIBUTING.md:30]: `alllowlist` or `allow-list` or `allow list`. Is the former used by any other projects?

File: api/envoy/config/filter/http/cache/v2alpha/cache.proto [hunk 3]
```diff
@@ -57,7 +57,7 @@ message CacheConfig {
   // contents of a response, as described by
   // https://httpwg.org/specs/rfc7234.html#caching.negotiated.responses.
   //
-  // During insertion, *allowed_vary_headers* acts as a whitelist: if a
+  // During insertion, *allowed_vary_headers* acts as a allowlist: if a
```
Comment (htuch) [inline api/envoy/config/filter/http/cache/v2alpha/cache.proto:60]: You might want to also add a rename annotation for `ip_white_list` in client_ssl_auth.proto.

Comment (mattklein123) [reply api/envoy/config/filter/http/cache/v2alpha/cache.proto:60]: Per above I will deal with client_ssl_auth and redis in a follow up.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mattklein123) [reply CONTRIBUTING.md:30]: From my unofficial research I see `allowlist` being used, but I don't feel very strongly about it.

File: bazel/foreign_cc/luajit.patch [hunk 4]
```diff
@@ -86,7 +86,7 @@ index 0000000..9c71271
 +      os.environ["HOST_CFLAGS"] = "-fno-sanitize=memory"
 +      os.environ["HOST_LDFLAGS"] = "-fno-sanitize=memory"
 +
-+    # Blacklist LuaJIT from ASAN for now.
++    # Remove LuaJIT from ASAN for now.
 +    # TODO(htuch): Remove this when https://github.com/envoyproxy/envoy/issues/6084 is resolved.
 +    if "ENVOY_CONFIG_ASAN" in os.environ:
 +      os.environ["TARGET_CFLAGS"] += " -fsanitize-blacklist=%s/com_github_luajit_luajit/clang-asan-blacklist.txt" % os.environ["PWD"]
```
Comment (lizan) [inline bazel/foreign_cc/luajit.patch:92]: change the file name too?

File: source/extensions/clusters/redis/redis_cluster.cc [hunk 5]
```diff
@@ -295,7 +295,7 @@ void RedisCluster::RedisDiscoverySession::onResponse(
   const uint32_t SlotRangeStart = 0;
   const uint32_t SlotRangeEnd = 1;
   const uint32_t SlotMaster = 2;
```
Comment (lizan) [inline source/extensions/clusters/redis/redis_cluster.cc:297]: `SlotPrimary`?

File: CONTRIBUTING.md (hunk 2 above)
Comment (ggreenway) [reply CONTRIBUTING.md:30]: One example from envoy: https://github.com/envoyproxy/envoy/blob/f87e68f8a7eef471b1e12b8ef1f662ab845356fb/api/envoy/extensions/filters/http/grpc_stats/v3/config.proto#L32

Comment (htuch) [reply CONTRIBUTING.md:30]: Yep, looking around it seems common enough, SGTM.

File: source/extensions/clusters/redis/redis_cluster.cc (hunk 5 above)
Comment (mattklein123) [reply source/extensions/clusters/redis/redis_cluster.cc:297]: I will do a follow up for redis. It's a larger change.

File: bazel/foreign_cc/luajit.patch (hunk 4 above)
Comment (mattklein123) [reply bazel/foreign_cc/luajit.patch:92]: I figured since the compiler flag has blacklist in it we should keep it but I can change it. Will do.

**Pull request merged**